### PR TITLE
[UI] Fix for showing disabled form controls in rule nodes in Angular-15

### PR DESCRIPTION
**Pull request by deaflynx**
## Pull Request description

Angular 15 migration caused UI bug for disabled rule node configuration controls. Issue details described [here](https://github.com/angular/angular/issues/48350#issuecomment-1405900515)

- Screenshot of the rule node configuration for tenant user without readOnly permissions:

Screenshot from 2023-06-12 18-04-17

- Screenshots for user with readOnly permissions:

Before:
Screenshot from 2023-06-12 18-04-02

After:
Screenshot from 2023-06-12 18-04-54

PE PR:
https://github.com/thingsboard/thingsboard-pe/pull/1809

## General checklist

- [ ] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [ ] [Labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels#about-labels) that classify your pull request have been added.
- [ ] The [milestone](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/about-milestones) is specified and corresponds to fix version.  
- [ ] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [ ] Description contains human-readable scope of changes.
- [ ] Description contains brief notes about what needs to be added to the documentation.
- [ ] No merge conflicts, commented blocks of code, code formatting issues.
- [ ] Changes are backward compatible or upgrade script is provided.
- [ ] Similar PR is opened for PE version to simplify merge. Crosslinks between PRs added. Required for internal contributors only.
  
## Front-End feature checklist

- [ ] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [ ] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [ ] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [ ] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [ ] If new dependency was added: the dependency tree is checked for conflicts.
- [ ] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [ ] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

**Comment by ashvayka** [comment]
Name and Description fields should be disabled as well.

File: ui-ngx/src/app/modules/home/pages/rulechain/rule-node-config.component.ts [hunk 1]
```diff
@@ -207,4 +207,15 @@ export class RuleNodeConfigComponent implements ControlValueAccessor, OnInit, On
       this.definedConfigComponent.validate();
     }
   }
+
+  private transformControls(configuration: RuleNodeConfiguration): any {
+    const result = {};
+    for (const [key, value] of Object.entries(configuration)) {
+      result[key] = {
+        value,
+        disabled: this.disabled
```
Comment (ashvayka) [inline ui-ngx/src/app/modules/home/pages/rulechain/rule-node-config.component.ts:216]: What if some inner components are disabled based on the logic specific to the particular rule node implementation? This implementation will work with primitive forms. But will not work with complex forms.

File: ui-ngx/src/app/modules/home/pages/rulechain/rule-node-config.component.ts [hunk 2]
```diff
@@ -148,7 +148,7 @@ export class RuleNodeConfigComponent implements ControlValueAccessor, OnInit, On
       this.changeSubscription = null;
     }
     if (this.definedConfigComponent) {
-      this.definedConfigComponent.configuration = this.configuration;
+      this.definedConfigComponent.configuration = this.transformControls(this.configuration);
```
Comment (vvlladd28) [inline ui-ngx/src/app/modules/home/pages/rulechain/rule-node-config.component.ts:151]: In definedConfigComponent need to add a new input value - disabled

File: ui-ngx/src/app/modules/home/pages/rulechain/rule-node-config.component.ts (hunk 1 above)
Comment (deaflynx) [reply ui-ngx/src/app/modules/home/pages/rulechain/rule-node-config.component.ts:216]: Ok, I will review and update pr

**Comment by vvlladd28** [comment]
Fixed in https://github.com/thingsboard/thingsboard/pull/9123

**Pull request closed without merging**